crates/volta-core/src/tool/package/manager.rs

```diff
@@ -11,72 +12,71 @@ use crate::fs::read_dir_eager;
 )]
 pub enum PackageManager {
     Npm,
+    Pnpm,
     Yarn,
 }
 
 impl PackageManager {
     /// Given the `package_root`, returns the directory where the source is stored for this
     /// package manager. This will include the top-level `node_modules`, where appropriate.
-    pub fn source_dir(self, package_root: PathBuf) -> PathBuf {
-        let mut path = self.source_root(package_root);
-        path.push("node_modules");
-
-        path
+    pub fn source_dir<P: AsRef<Path>>(self, package_root: P) -> PathBuf {
+        self.source_root(package_root).join("node_modules")
     }
 
     /// Given the `package_root`, returns the root of the source directory. This directory will
     /// contain the top-level `node-modules`
     #[cfg(unix)]
-    pub fn source_root(self, package_root: PathBuf) -> PathBuf {
-        // On Unix, the source is always within a `lib` subdirectory, with both npm and Yarn
-        let mut path = package_root;
-        path.push("lib");
-
-        path
+    pub fn source_root<P: AsRef<Path>>(self, package_root: P) -> PathBuf {
```

Comment:
Is there a change elsewhere that is requiring the switch to accepting an `AsRef<Path>` here? It's minor, but this approach requires an extra copy of the `Path` value to convert it into a `PathBuf`, which the original didn't.

Could we instead continue to use `PathBuf` even for the pnpm case?